operators/pkg/controller/elasticsearch/driver/downscale.go

```diff
@@ -0,0 +1,134 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package driver
+
+import (
+	"github.com/elastic/cloud-on-k8s/operators/pkg/apis/elasticsearch/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common/reconciler"
+	esclient "github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/client"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/label"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/migration"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/observer"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/reconcile"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/sset"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/version/zen1"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/version/zen2"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/k8s"
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+func (d *defaultDriver) HandleDownscale(
+	es v1alpha1.Elasticsearch,
+	expectedStatefulSets sset.StatefulSetList,
+	actualStatefulSets sset.StatefulSetList,
+	esClient esclient.Client,
+	observedState observer.State,
+	reconcileState *reconcile.State,
+) *reconciler.Results {
+	results := &reconciler.Results{}
+
+	// compute the list of nodes leaving the cluster, from which
+	// data should be migrated away
+	leavingNodes := []string{}
```

Comment:
```suggestion
var leavingNodes []string
```
?

Reply:
golangci-lint complains if I do that:

```
⟩ make lint
golangci-lint run
pkg/controller/elasticsearch/driver/downscale.go:34:2: Consider preallocating `leavingNodes` (prealloc)
        var leavingNodes []string
```

In general this warning makes sense since we can pre-allocate the slice to the correct size, but here I cannot know the size in advance 🤷‍♂

Reply:
Even the linter says it's only if you can tell how large it is in advance:
https://github.com/alexkohler/prealloc#purpose
> If the size of a slice is known at the time of its creation, it should be specified.
I don't remember the flag to tell the linter to quiet down though. It doesn't make that much of a difference though